### DISCUSSION TITLE
Implementing a moddable asset packing system in bevy

### DISCUSSION THREAD

**Discussion by ItsSunnyMonster** [bevyengine/bevy#13514] [Q&A]
I am making an asset packing system that supports loading multiple asset packs from a folder, where the user can add their own asset packs to override some files of the original asset pack. I think I can do this by implementing `AssetReader` to read from the asset packs. 

However, it would also mean that the user can add more packs while the game is running. When that happens, I have to update some metadata on the `AssetReader` (namely a `HashMap` or similar that stores which pack file to find each asset) and also reload all the assets that have been modified.

Is that possible in bevy? If so, could you please provide an outline of how that might be achieved? Thank you.

**Comment by Anatoly03** [comment]
Did you find a solution?  I'm thinking of having a "spritesheet" resource which can be extended by various rust modules.
